### PR TITLE
Add CLI endpoint to validate input files

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Try it like this:

```
spectre validate INPUT_FILE
```

In particular, this will be used in #4996 to validate input files before scheduling runs.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tools/ValidateInputFile.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+class InvalidInputFileError(Exception):
+    """Indicates that the input file cannot be parsed by the executable
+
+    The exception prints the 'message' by default. To print additional context,
+    catch the exception and print the 'render_context()' as well.
+
+    Attributes:
+      input_file_path: Path to the input file on disk.
+      lineno: Line number in the input file where the parse error occurred.
```
Comment (knelli2) [inline tools/ValidateInputFile.py]: I think `line_number` is clearer and it's not that much longer.

**Review by knelli2 — CHANGES_REQUESTED** [review]
I think you can drop the build dir commit because it's already taken care of in #5023. Then I believe these can be merged simultaneously rather than having to rebase one.

**Review by knelli2 — DISMISSED** [review]

**Review by knelli2 — CHANGES_REQUESTED** [review]
This test timed out a couple times, maybe because it takes a sec to get going? Probably should increase test timeout if possible

**Comment by knelli2** [comment]
Looks good, but I think you used black to format instead of yapf because of your other PR.

**Review by knelli2 — APPROVED** [review]

**Comment by nilsvu** [comment]
@kidder @wthrowe can one of you take this one please?

**Review by kidder — APPROVED** [review]

**Pull request merged**